tests/e2e/workloads/test_pod_from_registry.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+import logging
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import E2ETest, workloads
+from tests import helpers
+from ocs_ci.utility import templating
+from ocs_ci.ocs.exceptions import UnexpectedImage
+
+logger = logging.getLogger(__name__)
+
+
+@workloads
+@pytest.mark.polarion_id("OCS-2085")
+class TestRegistryImage(E2ETest):
+    """
+    Spin up a pod using custom image from openshift registry
+    """
+    pvc_size = 5
+
+    def test_run_pod_local_image(self, pvc_factory, pod_factory):
+        """
+        Run a pod with image backed by local registry
+        """
+        pvc_obj = pvc_factory(size=self.pvc_size)
+        image_obj = helpers.create_build_from_docker_image(
+            namespace=pvc_obj.namespace,
+            source_image_label='7',
+            image_name='fio',
+            install_package='fio'
+        )
+        image_id = image_obj.get().get('status').get('tags')[0].get('items')[0].get('dockerImageReference')
+        pod_dict = templating.load_yaml(constants.CSI_CEPHFS_POD_YAML)
+        pod_dict['spec']['containers'][0]['image'] = image_obj.resource_name
+        pod_dict['spec']['volumes'][0]['persistentVolumeClaim']['claimName'] = pvc_obj._name
+        pod_obj = pod_factory(pvc=pvc_obj, custom_data=pod_dict)
+        pod_image = pod_obj.get().get('status').get('containerStatuses')[0].get('imageID')
+        if image_id == pod_image:
```

Comment:
consider using assert, lines 37-40 could be replaced with:
```python
assert image_id == pod_image, f'pod uses different image {pod_image}'
logger.info(f'pod uses {image_id}')
```

Reply:
I made this change as per your suggestion.